CraftTweaker2-MC1120-Main/src/main/java/crafttweaker/mc1120/player/MCPlayer.java

```diff
@@ -1,279 +1,270 @@
+        return player.capabilities.isCreativeMode;
+    }
+    
+    @Override
+    public boolean isAdventure() {
+        return !player.capabilities.allowEdit;
+    }
+    
+    @Override
+    public void openBrowser(String url) {
+        if(player instanceof EntityPlayerMP) {
+            CraftTweaker.NETWORK.sendTo(new MessageOpenBrowser(url), (EntityPlayerMP) player);
+        }
+    }
+    
+    @Override
+    public void copyToClipboard(String value) {
+        if(player instanceof EntityPlayerMP) {
+            CraftTweaker.NETWORK.sendTo(new MessageCopyClipboard(value), (EntityPlayerMP) player);
+        }
+    }
+    
+    @Override
+    public boolean equals(Object other) {
+        return other != null && other.getClass() == this.getClass() && ((MCPlayer) other).player == player;
+        
+    }
+    
+    @Override
+    public int hashCode() {
+        int hash = 5;
+        hash = 23 * hash + (this.player != null ? this.player.hashCode() : 0);
+        return hash;
+    }
+    
+    @Override
+    public void give(IItemStack stack) {
+        ItemStack itemstack = CraftTweakerMC.getItemStack(stack).copy();
+        ItemHandlerHelper.giveItemToPlayer(player, itemstack);
+    }
+    
+    @Override
+    public void teleport(Position3f pos) {
+        player.setPosition(pos.getX(), pos.getY(), pos.getZ());
+    }
+    
+    @Override
+    public void changeDimension(int dimensionId) {
+        player.changeDimension(dimensionId);
+    }
+    
+    @Override
+    public int getScore() {
+        return player.getScore();
+    }
+    
+    @Override
+    public void addScore(int amount) {
+        player.addScore(amount);
+    }
+    
+    @Override
+    public void setScore(int amount) {
+        player.setScore(amount);
+    }
+    
+    @Override
+    public IFoodStats getFoodStats() {
+        return new MCFoodStats(player.getFoodStats());
+    }
+    
+    @Override
+    public void executeCommand(String rawCommand) {
+        MinecraftServer server = player.getServer();
+        if(server != null)
+            server.getCommandManager().executeCommand(player, rawCommand);
+    }
+    
+    @Override
+    public boolean isDamageDisabled() {
+        return player.capabilities.disableDamage;
+    }
+    
+    @Override
+    public void setDamageDisabled(boolean disabled) {
+        player.capabilities.disableDamage = disabled;
+    }
+    
+    @Override
+    public boolean canFly() {
+        return player.capabilities.allowFlying;
+    }
+    
+    @Override
+    public void setCanFly(boolean canFly) {
+        player.capabilities.allowFlying = canFly;
+    }
+    
+    @Override
+    public boolean canEdit() {
+        return player.capabilities.allowEdit;
+    }
+    
+    @Override
+    public void setCanEdit(boolean canEdit) {
+        player.capabilities.allowEdit = canEdit;
+    }
+
+    @Override
+    public String getUUID() {
+        return EntityPlayer.getUUID(player.getGameProfile()).toString().toLowerCase();

```

Comment:
Before, this was 
```java
    @Override	
    public String getUUID() {	
        UUID uuid = null;	
        if (player.getGameProfile() != null) {	
            uuid = EntityPlayer.getUUID(player.getGameProfile());	
        }	
        if(uuid == null) {	
            uuid = EntityPlayer.getOfflineUUID(getName());	
        }	
        if(uuid == null) {	
            CraftTweakerAPI.logError("Could not get UUID for player " + getName());	
            return "";	
        }	

        return uuid.toString().toLowerCase();	
    }
```

Is it safe to assume that getUUID will always work so we don't have to check the others?

Reply:
yes